app/controllers/products_controller.rb

```diff
@@ -0,0 +1,97 @@
+class ProductsController < ApplicationController
+  helper_method :current_user, :render_404, :require_login
+
+  before_action :find_product, only: [:show, :edit, :update, :destroy, :toggle_status]
+  
+  around_action :require_login, only: [:new, :create, :edit, :update, :destroy, :toggle_status], if: -> { !@login_user }
+  around_action :render_404, only: [:show, :edit, :update, :destroy, :toggle_status], if: -> { @product.nil? }
+  around_action :check_authorization, only: [:edit, :update, :destroy, :toggle_status], if: -> { @login_user && @login_user != @product.user }
+
+
+  def index 
+    @products = Product.where(active: true).order(:name).paginate(:page=>params[:page],:per_page=>15)
```

Comment:
you could consider during some of this work in a custom model method such as `active_sorted_products`.